compare/values_differ_any.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package compare
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var _ ValueComparer = valuesDifferAny{}
+
+type valuesDifferAny struct{}
+
+// CompareValues determines whether any value in the supplied values
+// is unique.
+func (v valuesDifferAny) CompareValues(values ...any) error {
+	if len(values) < 2 {
+		return nil
+	}
+
+	var val any
+
+	for i := 0; i < len(values); i++ {
```

Comment:
nit: I might be oversimplifying this, but can we just take the first element and compare it to each of the remaining elements? Rather than the nested loop.

Reply:
Thinking about this a little further, I'm not sure there's a use case for `valuesDifferAny`. In fact, I'm not sure there's a use case for the built-in values comparers except for `valuesDiffer` and `valuesSame`. The 3 built-in compare value state checks are for comparing values retrieved from state from sequential `terraform apply` runs (i.e., `CompareValue` which replaces `TestCheckResourceAttrPtr`), comparing pairs of values retrieved from state (i.e., `CompareValuePairs` which replaces `TestCheckResourceAttrPair`), and determining if a collection contains a value (i.e., `CompareValueCollection` which replaces `TestCheckTypeSetElemAttrPair`). In each of these cases, the use cases for the value comparers is to determine whether the sequence of values passed in differ or are the same.

Consequently, I'm going to remove `valuesDifferAll`, `valuesDifferAny`, and `valuesSameAny`.

Reply:
Ah okay, that makes sense. We could always come back and add more comparer implementations if necessary 👍🏻